ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezSessionState.java

```diff
@@ -310,7 +311,11 @@ protected void openInternal(final HiveConf conf, Collection<String> additionalFi
 
     setupSessionAcls(tezConfig, conf);
 
-    final TezClient session = TezClient.newBuilder("HIVE-" + sessionId, tezConfig)
+    String tezJobName = HiveConf.getVar(conf,ConfVars.HIVETEZJOBNAME); 
+    if (StringUtils.isBlank(tezJobName)){
+      tezJobName = String.format("HIVE-%s", sessionId);
```

Comment:
This is not quite what I was suggesting.  Right now, the format only happens for the default value.  it should be allowed for the user's customer value as well.  I'm not even sure I would worry about the `isBlank` check.  Just rely on the default value. of `HIVETEZJOBNAME`